qa/pull-tester/rpc-tests.sh

```diff
@@ -34,6 +34,7 @@ testScripts=(
     'maxblocksinflight.py'
     'invalidblockrequest.py'
     'rawtransactions.py'
+    'retarget-overflow.py'
```

Comment:
Ignore my suggestion to disable retarget on regtest, as testing retarget on
mainnet itself is not feasible.

Can we easily redefine it to not require larger arithmetic?

If not, switching everything to 512 and adding tests for it like this PR
does is probably the best option, as it's the only way to make sure the
code under test is the same as the one affecting mainnet.